packages/design-system-react-native/src/components/Checkbox/README.md

```diff
@@ -1,25 +1,60 @@
 # Checkbox
 
-`Checkbox` allows users to select one or more options from a set of choices
+`Checkbox` allows users to select one or more options from a set of choices.
+
+_Developer Note: This is a fully controlled component, requiring you to manage the state with `isSelected` and `onChange` props._
```

Comment:
Adding developer note that this is a fully controlled component although I don't think it's completely necessary seeing as that all instances of Checkboxes in both extension and mobile are controlled